apps/web/src/components/Menu/Menu.tsx

```diff
@@ -21,21 +21,23 @@ import {
 } from "@chakra-ui/react";
 import { useSelectNetwork, useSelectedNetwork } from "@umami/state";
 import { useRef } from "react";
+import { useNavigate } from "react-router";
 
 import { LogoutIcon } from "../../assets/icons";
 import { persistor } from "../../utils/persistor";
 
 export const Menu = () => {
   const selectNetwork = useSelectNetwork();
   const currentNetwork = useSelectedNetwork();
+  const navigate = useNavigate();
 
   const { isOpen, onOpen, onClose } = useDisclosure();
   const cancelRef = useRef(null);
 
   const handleLogout = () => {
     persistor.pause();
     localStorage.removeItem("persist:accounts");
-    window.location.replace("/");
+    navigate("/");
```

Comment:
with this, it's likely that the persistor will still be paused which is not exactly what we expect